docs/devel/kubemark-guide.md

```diff
@@ -0,0 +1,139 @@
+ready Kubemark cluster, script stores a kubeconfig file for talking to the Kubemark cluster in `test/kubemark/kubeconfig.loc`.
+
+Currently single HollowNode uses 50 millicores of CPU and ~60MB or memory, which taking into account default cluster addons and fluentD, allows running ~17.5 HollowNodes per
+core.
+
+#### Behind the scene details:
+
+Start-kubemark script does quite a lot of things:
+- Creates a master machine called hollow-cluster-master and PD for it (*uses gcloud, should be easy to do outside of GCE*)
+- Creates a firewall rule which opens port 443 on the master machine (*uses gcloud, should be easy to do outside of GCE*)
+- Builds a Docker image for HollowNode from the current repository and pushes it to the Docker repository (*GCR for us, using scripts from `cluster/gce/util.sh` - it may get
+tricky outside of GCE*)
+- Generates certificates and kubeconfig files, writes a kubeconfig locally to `test/kubemark/kubeconfig.loc` and creates a Secret which stores kubeconfig for HollowKubelet/
+HollowProxy use (*used gcloud to transfer files to Master, should be easy to do outside of GCE*).
+- Creates a ReplicationController for HollowNodes and starts them up. (*will work exactly the same everywhere as long as MASTER_IP will be populated correctly, but you’ll need
+to update docker image address if you’re not using GCR and default image name*)
+- Waits until all HollowNodes are in the Running phase (*will work exactly the same everywhere*)
+
+### Running e2e tests on Kubemark cluster
+
+To run standard e2e test on your Kubemark cluster created in the previous step you execute `test/kubemark/run-scalability-test.sh` script. It will configure ginkgo to
+use Kubemark cluster instead of something else and start a density test. This script should not need any changes to work on other cloud providers.
+
+If you want to run different test you just need to provide flags you want to be passed to `hack/ginkgo-e2e.sh` script, if you won't pass any flags to the script it will run Density '30 test.
+
+### Monitoring test execution and debugging problems
+Run-scalability-test prints the same output on Kubemark as on ordinary e2e cluster, but if you need to dig deeper you need to learn how to debug HollowNodes and how Master
+machine (currently) differs from the ordinary one.
+
+If you need to debug master machine you can do similar things as you do on your ordinary master. The difference between Kubemark setup and ordinary setup is that in Kubemark
+etcd is run as a plain docker container, and all master components are run as normal processes. There’s no Kubelet overseeing them. Logs are stored in exactly the same place,
+i.e. `/var/logs/` directory.
+
+To help you with debugging from inside the cluster startup script puts a `~/configure-kubectl.sh` script on the master. It downloads `gcloud` and `kubectl` tool and configures
```

Comment:
And possibly run it there - then you will be able to just log in and start using kubectl - it would be very convenient.